### PR TITLE
Retarget to netstandard2.0

### PR DESCRIPTION
**Pull request by Tratcher**
Surprisingly didn't have to revert any of the product code changes.
Compare to https://github.com/aspnet/HttpSysServer/commit/31001de66a20d9e056247b558f421afedcf1d0b9

### CODE REVIEW

File: build/dependencies.props [hunk 1]
```diff
@@ -3,7 +3,7 @@
     <AspNetCoreVersion>2.0.0-*</AspNetCoreVersion>
     <CoreFxVersion>4.3.0</CoreFxVersion>
```
Comment (JunTaoLuo) [inline build/dependencies.props]: Should be updated to 4.4.0-*

File: test/Microsoft.AspNetCore.Server.HttpSys.FunctionalTests/Listener/ResponseSendFileTests.cs [hunk 2]
```diff
@@ -362,7 +362,19 @@ public ResponseSendFileTests()
                 var writeTask = context.Response.SendFileAsync(AbsoluteFilePath, 0, null, cts.Token);
                 Assert.True(writeTask.IsCanceled);
                 context.Dispose();
-
+#if NET461
+                // .NET 4.5 HttpClient automatically retries a request if it does not get a response.
```
Comment (JunTaoLuo) [inline test/Microsoft.AspNetCore.Server.HttpSys.FunctionalTests/Listener/ResponseSendFileTests.cs]: I guess this is now .NET 4.6.1 HttpClient too

File: test/Microsoft.AspNetCore.Server.HttpSys.FunctionalTests/Listener/AuthenticationTests.cs [hunk 3]
```diff
@@ -41,7 +41,8 @@ public class AuthenticationTests
             }
         }
 
-        [ConditionalTheory(Skip = "HttpClientHandler issue (https://github.com/aspnet/ServerTests/issues/82).")]
+        [ConditionalTheory]
+        [FrameworkSkipCondition(RuntimeFrameworks.CoreCLR, SkipReason = "HttpClientHandler issue (https://github.com/aspnet/ServerTests/issues/82).")]
```
Comment (natemcmaster) [inline test/Microsoft.AspNetCore.Server.HttpSys.FunctionalTests/Listener/AuthenticationTests.cs]: Any idea if this is fixed? https://github.com/aspnet/ServerTests/issues/82 references https://github.com/dotnet/corefx/issues/5045 which is closed

Comment (Tratcher) [inline test/Microsoft.AspNetCore.Server.HttpSys.FunctionalTests/Listener/AuthenticationTests.cs]: No it wasn't, https://github.com/dotnet/corefx/issues/5045 was closed as won't fix for our scenario.

**Pull request merged**